lib/ansible/modules/cloud/amazon/iam_saml_federation.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: iam_saml_federation
+version_added: "2.3"
+short_description: maintain iam saml federation configuration.
+requires: [ boto3 ]
+description:
+    - Provides a mechanism to manage AWS IAM SAML Identity Federation providers (create/update/delete metadata). IAM Identity Federation allows you to easily enable a large number of users access to AWS resources without creating individual accounts in AWS.  More information on what an Identity Provider is and how to configure the appropriate IAM Roles is avaialble in the official AWS documentation (http://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_saml.html) 
+options:
+    name:
+        description:
+            - The name of the provider to create.  The value of this field is limited to characters consisting of upper and lowercase alphanumeric characters with no spaces. You can also include any of the following characters: =,.@-
```

Comment:
Please add `"` to each end of this line, the colon at the end is making YAML interpret this as an object, so the docs build fails.